policy/evaluator.go

```diff
@@ -297,22 +298,36 @@ func (eval *evaluator) Eval(vars interface{}) ([]ref.Val, error) {
 	}
 	decisions := []ref.Val{}
 	terms := eval.activation.terms
-	activation := &evaluatorActivation{
-		input:     varActivation,
-		terms:     terms,
-		memoTerms: make(map[string]ref.Val),
+	activation := evalActivationPool.Get().(*evaluatorActivation)
```

Comment:
nit: it might be nicer to wrap evalActivationPool.Get() and setup in a function call.